rust/lance/src/dataset/blob.rs

```diff
@@ -209,6 +210,54 @@ pub(super) async fn take_blobs(
     }
 }
 
+/// Internal function for taking blobs by row addresses (not row IDs).
+///
+/// This is used by `take_blobs_by_indices` which already has row addresses
+/// computed from row indices. Using this function bypasses the row ID index
+/// lookup, which is necessary when `enable_stable_row_ids=true` because
+/// row addresses are not the same as row IDs in that case.
+pub(super) async fn take_blobs_by_addresses(
```

Comment:
I think we should make this a public method which can be invoked by callers directly with python and java bindings. Originally we had take_blobs which takes row_ids and row_ids can be row address, or stable row ID if it is enabled. This assumes stable row ID is the next generation of row address that replaces its use cases completely. However, I think the situation has changed now, that we do want to use both stable row ID and row address depending on different use cases, so it makes sense now that we expose APIs for them respectively.